### ISSUE TITLE
Wsl die after laptop sleep and can't be reopened 

### ISSUE THREAD

**Issue by Ansuel** [microsoft/WSL#9301]
### Version

Windows: [10.0.25252.1010]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.79.1

### Distro Version

Ubuntu from store (with apt sources set to devel and package updated)

### Repro Steps

1. Open WSL cmd from Windows Terminal
2. Open explorer.exe on a WSL directory
3. Put laptop to sleep
4. Notice Windows Terminal window closed
5. Notice explorer.exe stuck on loading the WSL directory
6. Open Windows Terminal to open a new WSL instance
7. Notice WSL fail to load due to timeout error

### Expected Behavior

Windows doesn't close Windows Terminal window and WSL should not be killed after sleep.
WSL should be recoverable in any case even if Windows kill the WSL process.

### Actual Behavior

WSL can't be recovered after Windows kills it after sleep and a reboot is required.

**Comment by kallja** [comment]
I'm experiencing the same issue. I'm not sure if it happens every time the computer wakes from sleep or just most times, but this is definitely a problem.

WSL seems completely stuck. `wsl.exe` seems to hang indefinitely at most commands I've tried, including `wsl --status`. `wsl --help` completes successfully.

When the problem manifests, the `Windows Subsystem for Linux` service can not be stopped nor restarted from the Services GUI. Shutdown just pends indefinitely.

As a remedy to avoid reboot, killing the `wslservice` by running `taskkill /F /pid <WSL_SERVICE_PID>` in an administrative shell causes the service to recover and WSL to be usable once again.

**Comment by Ansuel** [comment]
@kallja do you have by chance docker desktop installed? What image are you using? 
Do you have explorer window open in $wsl directory while the problem arise?

Wasn't aware of the taskkill procedure... That is useful for additional debug log. Nice!

(btw wsl --shutdown also stuck)

**Comment by Ansuel** [comment]
@OneBlue I finally found time to make a dump with the problem repro
here is the dump link https://we.tl/t-mTYroAcEj7

**Comment by Ansuel** [comment]
@kallja can confirm that by using the taskkill workaround wsl function is recovered. The bug can be repro by simply putting the pc to sleep back again...

**Comment by noamzbr** [comment]
Having the exact same issue, except that `taskkill /F /pid <WSL_SERVICE_PID>` also doesn't work... Didn't find any fix so far apart from a full restart.

**Comment by Ansuel** [comment]
just open task manager details tab... find wslservice and kill... reopen the related wsl instance. Also can you provide info of running kernel, win build and other?

**Comment by Ansuel** [comment]
think you should open another issue

**Comment by JustThat70** [comment]
> think you should open another issue

Perhaps not as they were just closed as being duplicates of this one even though they have somewhat different symptoms, and in my case, a different version of WSL

**Comment by JustThat70** [comment]
I've learned that if I perform a `wsl --shutdown` when I am done and before the PC goes to sleep or hibernates, that wsl will launch after waking

**Comment by kallja** [comment]
Sorry for the delay in answering.

> @kallja do you have by chance docker desktop installed? What image are you using?

I do have Docker Desktop installed. When the computer wakes from sleep, WSL is dead whether there were running Docker containers or not. That doesn't seem to be a contributing factor.

> Do you have explorer window open in $wsl directory while the problem arise?
No. Nowadays it happens pretty much always when my computer wakes from sleep.

**Comment by Ansuel** [comment]
Can confirm this is still present in latest dev build.

**Comment by pfz** [comment]
I confirm that too. After hibernation S0/sleep/hybrid, whichever, gui killed, wsl still there but not responsive. `wsl -d mydistro` hangs forever. `wsl --shutdown` doesn't get stuck anymore **all the time** though. Getting logs that could have occured after wake is problematic, nothing there, and system wsl is not reachable either.

**Comment by drummonds** [comment]
I have this problem as well and am wondering if it is to do with systemmd being turned on because I had problems with date time sync which gave me problems with hugo generating pages (as it thought they were in the future)

**Comment by runfaj** [comment]
Been having something _kindof_ similar. Essentially the container is still running and I can still run commands and interact with it inside the container. However, any external commands (like docker desktop, any from vscode, etc) basically just die and never work. The ONLY way to fix is to completely shutdown WSL and restart it. This is definitely not ideal. It seems there is some intefacing bug with the running wsl instance that needs to be resolved.

**Comment by msokolovsky-appulate** [comment]
hate this bug.
```WSL version: 2.1.3.0
Kernel version: 5.15.146.1-2
WSLg version: 1.0.60
MSRDC version: 1.2.5105
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.19045.4651
```
workaround:
```taskkill /f /im wsl.exe

taskkill -IM "wslservice.exe" /F

# Stop the WSL2 service
Stop-Service -Name "LxssManager"

# Start the WSL2 service
Start-Service -Name "LxssManager"```

**Comment by eads** [comment]
This is affecting me on a ~3 year old Lenovo Legion. 

I found the PID with `ps` and then used `taskkill /F /pid <PID>` but now all my terminals say:

```
[process exited with code 1 (0x00000001)]
You can now close this terminal with Ctrl+D, or press Enter to restart.
```

Pressing Enter just dumps me back into a new shell. 

It's not clear to me what sleep conditions trigger this; I went days without being affected this most recent time but this morning it happened on wake.
